crates/runtime/src/dataconnector/odbc.rs

```diff
@@ -86,7 +95,23 @@ impl From<ODBCDriver> for Option<Arc<dyn Dialect + Send + Sync>> {
             ODBCDriver::MySql => Some(Arc::new(MySqlDialect {})),
             ODBCDriver::PostgreSql => Some(Arc::new(PostgreSqlDialect {})),
             ODBCDriver::Sqlite => Some(Arc::new(SqliteDialect {})),
-            ODBCDriver::Unknown => None,
+            ODBCDriver::Unknown => Some(Arc::new(DefaultDialect {})),
+        }
+    }
+}
+
+impl TryFrom<SQLDialectParam> for Option<Arc<dyn Dialect + Send + Sync>> {
```

Comment:
in the spiceai fork, the Dialect should be already Send + Sync?

Reply:
it's because of the bounds included in the `with_dialect` function in `datafusion-table-providers`: https://github.com/datafusion-contrib/datafusion-table-providers/blob/ab9457dc8aec7ebc392e22efe4176f11b47d369a/src/sql/sql_provider_datafusion/mod.rs#L107

You're correct though, that `Dialect` has `Dialect: Send + Sync` so it should already implicitly be Send + Sync.